### ISSUE TITLE
E3047 Cpu '256' is not compatible with memory '1GB' because a single space is required before 'GB' for Cpu '256'

### ISSUE THREAD

**Issue by sir-sigurd** [aws-cloudformation/cfn-lint#3879]
### CloudFormation Lint Version

cfn-lint 1.22.1

### What operating system are you using?

Mac

### Describe the bug

using cfn-lint on repro template results in 
```
E3047 Cpu '256' is not compatible with memory '1GB'
```
which is valid combination

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-taskdefinition.html#cfn-ecs-taskdefinition-cpu

> 256 (.25 vCPU) - Available memory values: 512 (0.5 GB), 1024 (1 GB), 2048 (2 GB)

I think this problem exists because for CPU 256 a single space is required before 'GB'

https://github.com/aws-cloudformation/cfn-lint/blob/310beb4e579b18fc531f5a0dfff54d5aa3cb88b7/src/cfnlint/data/schemas/extensions/aws_ecs_taskdefinition/fargate_cpu_memory.json#L55-L57

unlike other CPU configuration

https://github.com/aws-cloudformation/cfn-lint/blob/310beb4e579b18fc531f5a0dfff54d5aa3cb88b7/src/cfnlint/data/schemas/extensions/aws_ecs_taskdefinition/fargate_cpu_memory.json#L86

### Expected behavior

not raise error for repro template

### Reproduction template

```yaml
Resources:
 S3ProxyTaskDefinition:
    Type: AWS::ECS::TaskDefinition
    Properties:
      RequiresCompatibilities:
        - FARGATE
      Cpu: '256'
      Memory: 1GB
      ContainerDefinitions:
        - Name: s3-nginx
          Image: nginx
```
